core/src/main/java/io/questdb/cutlass/text/TextImportTask.java

```diff
@@ -0,0 +1,995 @@
+            }
+        }
+
+        private void mergePartitionIndexAndImportData(final FilesFacade ff,
+                                                      final Path partitionPath,
+                                                      final TextLexer lexer) throws TextException {
+            try (DirectLongList mergeIndexes = new DirectLongList(64, MemoryTag.NATIVE_DEFAULT)) {
+                partitionPath.slash$();
+                int partitionLen = partitionPath.length();
+
+                long mergedIndexSize = openIndexChunks(ff, partitionPath, mergeIndexes, partitionLen);
+                long address = -1;
+                long fd = -1;
+
+                try {
+                    final int indexesCount = (int) mergeIndexes.size() / 2;
+                    partitionPath.trimTo(partitionLen);
+                    partitionPath.concat(CsvFileIndexer.INDEX_FILE_NAME).$();
+
+                    fd = TableUtils.openFileRWOrFail(ff, partitionPath, CairoConfiguration.O_NONE);
+                    address = TableUtils.mapRW(ff, fd, mergedIndexSize, MemoryTag.MMAP_PARALLEL_IMPORT);
+                    ff.close(fd);
+                    fd = -1;
+
+                    final long merged;
+                    try {
+                        merged = Vect.mergeLongIndexesAscExt(mergeIndexes.getAddress(), indexesCount, address);
+                    } finally {
+                        //release chunk memory early
+                        //if there's only one chunk we didn't have to merge, can use it and have to release after loading data
+                        if (mergeIndexes.size() > 2) {
+                            unmap(ff, mergeIndexes);
+                        }
+                    }
+
+                    importPartitionData(lexer, merged, mergedIndexSize);
+                } finally {
+                    if (fd > -1) {
+                        ff.close(fd);
+                    }
+                    if (address != -1) {
+                        ff.munmap(address, mergedIndexSize, MemoryTag.MMAP_PARALLEL_IMPORT);
+                    }
+                    unmap(ff, mergeIndexes);
+                }
+            }
+        }
+
+        private void unmap(FilesFacade ff, DirectLongList mergeIndexes) {
+            for (long i = 0, sz = mergeIndexes.size() / 2; i < sz; i++) {
+                final long addr = mergeIndexes.get(2 * i);
+                final long size = mergeIndexes.get(2 * i + 1) * CsvFileIndexer.INDEX_ENTRY_SIZE;
+                ff.munmap(addr, size, MemoryTag.MMAP_PARALLEL_IMPORT);
+            }
+            mergeIndexes.clear();
+        }
+
+        private boolean onField(long offset, final DirectByteCharSequence dbcs, TableWriter.Row w, int fieldIndex, DirectCharSink utf8Sink)
+                throws TextException {
+            TypeAdapter type = this.types.getQuick(fieldIndex);
+            try {
+                type.write(w, fieldIndex, dbcs, utf8Sink);
+            } catch (Exception ignore) {
```

Comment:
better be `Throwable`

Reply:
Would it make sense to treat OOM the same as parse errors ?

Reply:
I'm skeptical about this change because it could hide serious runtime errors when atomicity is skip_col or skip_row  instead of propagating . 
Error skipping is only meant to handle bad data conversions .